src/troute-routing/troute/routing/diffusive_utils.py

```diff
@@ -940,13 +991,98 @@ def fp_crosswalk_map(
                         jcol = jcol + 3   
     
     return crosswalk_nrow, crosswalk_ncol, crosswalk_g 
+
+def fp_coastal_boundary_input_map(
+    tw,
+    coastal_boundary_depth_df, 
+    nrch_g,
+    t0,
+    #nsteps,
+    dt_db_g,
+    t0_g,
+    tfin_g):
+    """
+    Data assimilatoin data mapping between Python and Fortran
+    Parameters
+    ----------
+    tw -- (int) Tailwater segment ID
+    coastal_boundary_depth_df -- (DataFrame) coastal boundary depth data at one hour time steps
+    t0              -- (datetime) initial date
+    #nsteps          -- (int) number of simulation time steps
+    dt_db_g         -- (int)   coastal boundary input data timestep in sec
+    t0_g            -- (float) diffusive model's initial simulation time [hr] (by default, zero)
+    tfin_g          -- (float) diffusive model's final simulation time [hr] 
+    Returns
+    -------
+    dsbd_option     -- (int) 1 or 2 for coastal boundary depth data or normal depth data, respectively
+    nts_db_g        -- (int) number of coastal boundary input data timesteps
+    dbcd_g          -- (float) coastal boundary input data time series [m]
+    """
+    
+    nts_db_g  = int((tfin_g - t0_g) * 3600.0 / dt_db_g) + 1  # include initial time 0 to the final time    
```

Comment:
is 3600 a good assumption?  Or should the be paramaterized?

Reply:
3600 is a generic value for converting hour to sec.